lightning/src/ln/channelmanager.rs

```diff
@@ -2135,13 +2159,31 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>
 				});
 			},
 			HTLCSource::PreviousHopData(hop_data) => {
+				let prev_hop_data_node_id = hop_data.counterparty_node_id;
+				let prev_outpoint = hop_data.outpoint;
 				if let Err((counterparty_node_id, err)) = match self.claim_funds_from_hop(&mut channel_state_lock, hop_data, payment_preimage) {
 					Ok(()) => Ok(()),
 					Err(None) => {
-						// TODO: There is probably a channel monitor somewhere that needs to
-						// learn the preimage as the channel already hit the chain and that's
-						// why it's missing.
-						Ok(())
+						let preimage_update = ChannelMonitorUpdate {
+							update_id: CLOSED_CHANNEL_UPDATE_ID,
+							updates: vec![ChannelMonitorUpdateStep::PaymentPreimage {
+								payment_preimage: payment_preimage.clone(),
+							}],
+						};
+						// We update the ChannelMonitor on the backward link, after
+						// receiving an offchain preimage event from the forward link (the
+						// event being update_fulfill_htlc).
+						match self.chain_monitor.update_channel(prev_outpoint, preimage_update) {
+							Err(ChannelMonitorUpdateErr::PermanentFailure) => {
+								let err_msg = MsgHandleErrInternal::send_err_msg_no_close("Failed to update channel with preimage provided from forward link".to_owned(), prev_outpoint.to_channel_id());
```

Comment:
Why send an error message to the peer? We're probably not even connected to them anymore and the channel is closed - I dont think we need to generate an Err here so much as we need to log a critical error and be sad for the user who lost their monitor.

(and, if we drop it, can we drop the new `counterparty_node_id` field(s)?)

Reply:
True, that didn't make sense. Removed